query/src/main/scala/filodb/query/exec/MultiSchemaPartitionsExec.scala

```diff
@@ -53,7 +56,15 @@ final case class MultiSchemaPartitionsExec(queryContext: QueryContext,
     // clear stats since previous call to lookupPartitions set the stat with metric name that has suffix
     querySession.queryStats.clear()
     val lookupRes = source.lookupPartitions(dataset, partMethod, chunkMethod, querySession)
-    (lookupRes, Some(columnName))
+    // Accept only if lookupResult schema has a data column with name as the requested columnName.
+    // This is true generally for histogram queries where we rewrite the query to remove _sum, _count suffix.
+    // Check is needed since summaries are not columnized, and we don't want the rewrite to happen for summaries.
+    // If column does not exist, return original lookup result and don't change the query
+    val schemas = source.schemas(datasetRef).get
+    if (lookupRes.firstSchemaId.exists(s => schemas.apply(s).data.columns.exists(_.name == columnName)))
```

Comment:
Not a blocker since we already do two full lookups: 

Curious if we can avoid the second "full" scan  (only scan for 1 or two ts) ? This will allow for the check before scanning the index for all entries in the 2nd lookup ?

Reply:
Fair suggestion. This requires a more involved change, best done in a different PR on a need basis.